### PR TITLE
Issue 43: Improve unit test coverage for Bookkeeper Operator 

### LINKED ISSUE #43: Improve unit test coverage for Bookkeeper Operator
### Description
Current code coverage is < 50%. Low code coverage hinders with robust testing of changes and and does not allow us to fail fast.

### Importance
must have

### Suggestions for an improvement
Add more tests to cover all possible code paths such that we get better code coverage.

### PR DESCRIPTION
**Pull request by anishakj**
Signed-off-by: anisha.kj <anisha.kj@dell.com>

### Change log description

Increased the coverage of the packages v1alpha1 and util

### Purpose of the change

 Fixes #43

### What the code does

Added  unit tests.

### How to verify it

Running make test should pass

### CODE REVIEW

File: pkg/util/zookeeper_util_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		var err error
+		BeforeEach(func() {
+			bk = &v1alpha1.BookkeeperCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			bk.WithDefaults()
+			err = DeleteAllZnodes(bk, "bookie")
+			log.Printf("%v", err)
```
Comment (Prabhaker24) [inline pkg/util/zookeeper_util_test.go:33]: you can remove this printf()

Comment (anishakj) [reply pkg/util/zookeeper_util_test.go]: Done

File: pkg/apis/bookkeeper/v1alpha1/status_test.go [hunk 2]
```diff
@@ -72,13 +141,170 @@ var _ = Describe("BookkeeperCluster Status", func() {
...
+			})
+			It("Checking GetlastCondition function and It should return RollbackCondition as cluster in Rollback state", func() {
+				condition := bk.Status.GetLastCondition()
+				Ω(string(condition.Type)).To(Equal(v1alpha1.ClusterConditionRollback))
+			})
+			It("Checking ClusterInUpgradeFailedOrRollbackState and It should return true", func() {
+				Ω(bk.Status.IsClusterInUpgradeFailedOrRollbackState()).To(Equal(true))
+			})
+
+			Context("set pods rollback condition to be false", func() {
```
Comment (SrishT) [inline pkg/apis/bookkeeper/v1alpha1/status_test.go:291]: Not aligned similar to the other Contexts

File: pkg/util/bookkeepercluster_test.go [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			bk.WithDefaults()
+			image = BookkeeperImage(bk)
+		})
+
+		It("should return correct image", func() {
+			Ω(image).To(Equal("pravega/bookkeeper:0.6.1"))
```
Comment (SrishT) [inline pkg/util/bookkeepercluster_test.go:215]: This default `pravega/bookkeeper:0.6.1` might change and we would need to alter this test every time

File: pkg/util/k8sutil_test.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+				},
+			}
+			bk.WithDefaults()
+			s := scheme.Scheme
+			s.AddKnownTypes(v1alpha1.SchemeGroupVersion, bk)
+			s.AddKnownTypes(v1alpha1.SchemeGroupVersion, bk1)
+			client = fake.NewFakeClient(bk)
+			err = WaitForClusterToTerminate(client, bk)
+		})
+		It("should  be nil", func() {
```
Comment (SrishT) [inline pkg/util/k8sutil_test.go:149]: Could we try to get the bk cluster and check whether it has actually terminated here?

File: pkg/apis/bookkeeper/v1alpha1/status_test.go [hunk 5]
```diff
@@ -72,13 +141,170 @@ var _ = Describe("BookkeeperCluster Status", func() {
...
 				Ω(condition.LastTransitionTime).NotTo(Equal(""))
 			})
 		})
+		Context("set pod upgrade condition to be true", func() {
+			BeforeEach(func() {
+				bk.Status.SetUpgradingConditionFalse()
+				bk.Status.SetUpgradingConditionTrue(" ", " ")
+				bk.Status.UpdateProgress("UpdatingBookkeeperReason", "0")
+			})
+			It("should have pod upgrade condition with true status", func() {
```
[the remark this answers is not in the document]
Comment (anishakj) [reply pkg/apis/bookkeeper/v1alpha1/status_test.go:160]: Done

File: pkg/apis/bookkeeper/v1alpha1/status_test.go (hunk 2 above)
Comment (anishakj) [reply pkg/apis/bookkeeper/v1alpha1/status_test.go]: Fixed

File: pkg/util/bookkeepercluster_test.go (hunk 3 above)
Comment (anishakj) [reply pkg/util/bookkeepercluster_test.go:218]: Fixed by setting image in version field

File: pkg/util/k8sutil_test.go (hunk 4 above)
Comment (anishakj) [reply pkg/util/k8sutil_test.go:149]: This is not feasible, even if we create cluster with some pods, we cant terminate the pods while the function is running

**Pull request merged**